contracts/libraries/VaultLifecycle.sol

```diff
@@ -142,7 +142,7 @@ library VaultLifecycle {
 
         newPricePerShare = currentSupply > 0
             ? singleShare.mul(roundStartBalance).div(currentSupply)
-            : singleShare;
+            : vaultParams.initialSharePrice;
```

Comment:
I like this approach because it moves the storage access of `vaultParams.initialSharePrice` to `rollToNextOption`, which means it won't be in the deposit/withdrawal flow